draft-private-access-tokens.md

```diff
@@ -416,15 +440,91 @@ Issuance assumes the Client has the following information, derived from a given
+start and end time for each Issuer requested by the Client. Associated with this window,
+the Mediator determines a CLIENT_GROUP_ID value for the Client-Issuer pair, which
+changes every time the policy window elapses. The Mediator additionally maintains
+a counter of successful tokens issued within the policy window for each ANON_ORIGIN_ID
+requested by the Client, as well as a GROUP_ORIGIN_ID that it received for each
+ANON_ORIGIN_ID.
+
+The Mediator assigns CLIENT_GROUP_ID values to Clients in way that segments them into
+groups of roughly equal size. For example, at any time, Clients could be spread
+across 10 different CLIENT_GROUP_ID values. The specific values for the CLIENT_GROUP_ID
+are integers determined by the Mediator, and is not reused across different
+policy windows in a short period of time. One simple mechanism would be to have a counter
+for groups, up to the maximum value of uint64, before wrapping around.
+
+A specific CLIENT_GROUP_ID can be in one of three states:
+
+- Open: Clients can be assigned this CLIENT_GROUP_ID upon first access to a given
+Issuer for their specific policy window. This state lasts exactly the number
+of seconds in ISSUER_POLICY_WINDOW.
+- Draining: Clients can no longer be assigned this CLIENT_GROUP_ID for new access,
+but some Clients are still using this CLIENT_GROUP_ID. This state also lasts exactly
+the number of seconds in ISSUER_POLICY_WINDOW.
+- Closed: No Clients are using this CLIENT_GROUP_ID, and no new Clients can have
+have it assigned.
+
+Any particular CLIENT_GROUP_ID MUST only be used for a maximum of twice the
+ISSUER_POLICY_WINDOW, in seconds. This is the combination of the open and draining
+states.
+
+### Issuer State {#issuer-state}
+
+Issuers are expected to have the private key that corresponds to ISSUER_NAME_KEY,
+which allows them to decrypt the ORIGIN_NAME values in requests.
+
+Issuers also need to know the current ORIGIN_TOKEN_KEY public key and corresponding
+private key, for each ORIGIN_NAME that is served by the Issuer.
+
+During issuance, Issuers help Mediators validate requests by provding a GROUP_ORIGIN_ID
+in responses that is derived from a pair of ORIGIN_NAME and CLIENT_GROUP_ID values.
+
+GROUP_ORIGIN_ID MUST be a stable and unpredictable 32-byte value computed by the Issuer.
+Issuers MUST NOT change this value across token requests for the same ORIGIN_NAME and
+CLIENT_GROUP_ID.
+
+One possible mechanism for generating GROUP_ORIGIN_ID is for the Issuer to compute a PRF
+keyed by a per-Issuer secret (issuer_secret) over the ORIGIN_NAME and CLIENT_GROUP_ID,
+e.g., GROUP_ORIGIN_ID = HKDF(secret=issuer_secret, salt=CLIENT_GROUP_ID, info=ORIGIN_NAME).
+
+Issuers also can keep state about previous CLIENT_GROUP_ID values they have seen from
+a specific Mediator. Issuers SHOULD validate that a give CLIENT_GROUP_ID value is only
+used for at most twice the length of the ISSUER_POLICY_WINDOW. If a CLIENT_GROUP_ID
```

Comment:
Over what period of time? Meaning, when does the Issuer stop tracking the CLIENT_GROUP_ID? This should be the same time as the Mediator is prohibited from reusing it -- "ten times the policy window".

Reply:
Clarified, and gave 10x policy window as the minimum